workers/go/kitchensink/kitchen_sink.go

```diff
@@ -116,25 +124,61 @@ func handleAction(
 		if child.WorkflowType != "" {
 			childType = child.WorkflowType
 		}
-		err := workflow.ExecuteChildWorkflow(ctx, childType, child.GetInput()).Get(ctx, nil)
+		err := withAwaitableChoice(ctx, func(ctx workflow.Context) workflow.Future {
+			return workflow.ExecuteChildWorkflow(ctx, childType, child.GetInput()[0])
+		}, child.AwaitableChoice)
+		return nil, err
+	} else if patch := action.GetSetPatchMarker(); patch != nil {
+		if workflow.GetVersion(ctx, patch.GetPatchId(), workflow.DefaultVersion, 1) == 1 {
+			return handleAction(ctx, workflowState, patch.GetInnerAction())
+		}
+	} else if setWfState := action.GetSetWorkflowState(); setWfState != nil {
+		workflowState = setWfState
```

Comment:
That's in an `else` clause. This local variable assignment (even though the variable is defined in parameter) is effectively the last thing run in this function. If you want this assignment to apply to what the pointer is pointing to, I think you need:

```suggestion
		*workflowState = *setWfState
```

Reply:
Yeah, duh, that makes sense. I assumed it must've somehow been doing the right thing still because nothing was hanging but I guess I somehow never tripped a test that had a set and then await.

Deref on both sides complains about copying a mutex inside the proto thing. So I'll have to clone it (which is also annoying because of the type casting) or have a double pointer so I can just reassign the real pointer.

Reply:
Double pointer is the simplest option.

Reply:
Yeah, I considered changing to the struct, I should just do that.